lib/pbench/server/cache_manager.py

```diff
@@ -240,12 +243,16 @@ def extract(tarball_path: Path, path: str) -> str:
             The named file as a string
         """
         try:
-            return tarfile.open(tarball_path, "r:*").extractfile(path).read().decode()
+            tar = tarfile.open(tarball_path, "r:*")
         except Exception as exc:
             raise MetadataError(tarball_path, exc)
+        try:
+            return tar.extractfile(path).read().decode()
+        except Exception:
+            return None
 
     @staticmethod
-    def _get_metadata(tarball_path: Path) -> JSONOBJECT:
+    def _get_metadata(tarball_path: Path) -> Optional[JSONOBJECT]:
```

Comment:
Either the docstring description or the "Returns" section (or both) should probably be updated to reflect this change.